Source/swiftlint/LintCommand.swift

```diff
@@ -167,14 +168,9 @@ struct LintOptions: OptionsType {
     let strict: Bool
     let useScriptInputFiles: Bool
 
-    static func create(path: String)(useSTDIN: Bool)(configurationFile: String)(strict: Bool)
-        (useScriptInputFiles: Bool) -> LintOptions {
-        return LintOptions(path: path, useSTDIN: useSTDIN, configurationFile: configurationFile,
-            strict: strict, useScriptInputFiles: useScriptInputFiles)
-    }
-
     static func evaluate(mode: CommandMode) -> Result<LintOptions, CommandantError<()>> {
-        return create
+        let curriedInit = curry(self.init)
```

Comment:
has to be split off into its own function to avoid crippling the compiler due to complex expression evaluation.